support-workers/src/main/scala/com/gu/support/workers/lambdas/CreateZuoraSubscription.scala

```diff
@@ -70,19 +73,26 @@ object CreateZuoraSubscription {
 
   import com.gu.FutureLogging._
 
-  def createSubscription(
-    state: CreateZuoraSubscriptionState,
-    requestInfo: RequestInfo,
+  case class Impure(
```

Comment:
I wasn't sure what to call this, but this function `createSubscription` is the one that I'm going to rethink a lot in part 2.  I added it initially because there was an error about there being more than 8 parameters (which is a fair point)
But I think the refactor proper deserves a separate PR to avoid confusion.

Reply:
I'm not a fan of introducing concepts like pure and impure at a class level, it seems quite FP jargony to me and doesn't really say anything about what it contains from an application logic point of view.

An alternative approach could be to group all the services into a wrapper class `SubscriptionServices` or similar and I'd also love to get rid of one of `today` and `now` as it seems redundant to pass both.

Reply:
agree, in the part 2 refactor I realised it makes more sense for code clarity to group things by their lifecycle than whether they are pure or impure.
So first the things that come from the environment/config that are the same for every execution, and then the things that come from the State/input that vary per "run".
Then you can concentrate on the business logic separately from the plumbing.
I think this is pretty much what you suggested with a new SubscriptionServices class.
Ideally any extra structure should add meaning to the code, rather than obscure it.

Regarding the today and now, I agree and there must be a way to convert them so I'll have a quick look.

Reply:
I have managed to get rid of the today/now separation.  I have got rid of the Impure class completely and made it normal class parameters.
Have a look and see what you think.